src/Chainweb/BlockHeaderDB/PruneForks.hs

```diff
@@ -189,22 +189,44 @@ pruneForks_ logg cdb mar mir callback = do
       else
         withReverseHeaderStream cdb (mar - 1) mir
             $ S.foldM_ go (return (pivots, int (_getMaxRank mar), 0)) (\(_,_,!n) -> evaluate n)
+            . progress 200000 reportProgress
 
   where
+    ver = _chainwebVersion cdb
+    cid = _chainId cdb
+    reportProgress i a = logg Info
+        $ "inspected " <> sshow i
+        <> " block headers. Current height "
+        <> sshow (_blockHeight a)
+
     go :: ([BlockHash], BlockHeight, Int) -> BlockHeader -> IO ([BlockHash], BlockHeight, Int)
     go ([], _, _) cur = throwM $ InternalInvariantViolation
-        $ "PrunForks.pruneForks_: no pivots left at block " <> encodeToText (ObjectEncoded cur)
+        $ "PruneForks.pruneForks_: no pivots left at block " <> encodeToText (ObjectEncoded cur)
     go (!pivots, !prevHeight, !n) !cur
 
         -- This checks some structural consistency. It's not a comprehensive
         -- check. Comprehensive checks on various levels are availabe through
         -- callbacks that are offered in the module "Chainweb.Chainweb.PruneChainDatabase"
         | prevHeight /= curHeight && prevHeight /= curHeight + 1 =
             throwM $ InternalInvariantViolation
-                $ "PrunForks.pruneForks_: detected a corrupted database. Some block headers are missing"
+                $ "PruneForks.pruneForks_: detected a corrupted database. Some block headers are missing"
+                <> ". Current pivots: " <> encodeToText pivots
+                <> ". Current header: " <> encodeToText (ObjectEncoded cur)
+                <> ". Previous height: " <> sshow prevHeight
+        | _blockChainwebVersion  cur /= ver =
```

Comment:
Unfortunately, this may actually trigger on all dbs that originated from a bootstrap node. It seems that for some reason those dbs contain development genesis headers -- which is something that we should fix. (I think, this can happen if a node is started with the wrong chainweb version and a fix should prevent that a non-empty db is initialized with genesis headers that do not match the existing headers in the database)